Adding local ToListAsync method in Copilot Chat

teresaqhoang opened the pull request:
### Motivation and Context
<!-- Thank you for your contribution to the semantic-kernel repo!
Please help reviewers and future users, providing the following information:
  1. Why is this change required?
  2. What problem does it solve?
  3. What scenario does it contribute to?
  4. If it fixes an open issue, please link to the issue here.
-->
The `System.Linq.Async` package reference was removed from the core libraries so that everyone using SK wasn't forced to use it as well. This adds a helper `ToListAsync` method to enable async conversion of `IAsyncEnumerable<T>` to `List<T>` object.

### Description
<!-- Describe your changes, the overall approach, the underlying design.
     These notes will help understanding how your code works. Thanks! -->
Was seeing the following error when using local project references, so this is an effort to future proof any SK nuget updates

`'IAsyncEnumerable<MemoryQueryResult>' does not contain a definition for 'ToListAsync' and no accessible extension method 'ToListAsync' accepting a first argument of type 'IAsyncEnumerable<MemoryQueryResult>' could be found (are you missing a using directive or an assembly reference?)  `

### Contribution Checklist
<!-- Before submitting this PR, please make sure: -->
- [X] The code builds clean without any errors or warnings
- [X] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [X] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

adrianwyatt commented on samples/apps/copilot-chat-app/webapi/CopilotChatWebApi.csproj:
```diff
@@ -23,6 +23,7 @@
     <PackageReference Include="Microsoft.Identity.Web" Version="2.11.0" />
     <PackageReference Include="PdfPig" Version="0.1.8-alpha-20230423-3898f" />
     <PackageReference Include="Swashbuckle.AspNetCore" Version="6.5.0" />
+    <PackageReference Include="System.Linq.Async" Version="6.0.1" />
```
since it's non-obvious as to why this is here, could you add a comment in the project file explaining?

stephentoub commented:
If all you need is ToListAsync, it's also straightforward to write one's self:
```C#
internal static List<T> ToListAsync<T>(this IAsyncEnumerable<T> source)
{
    var result = new List<T>();
    await foreach (var item in source.ConfigureAwait(false)) result.Add(item);
    return result;
}
```

The `System.Linq.Async` package reference was removed from the core libraries so that everyone using SK wasn't forced to use it as well.

teresaqhoang commented:
> If all you need is ToListAsync, it's also straightforward to write one's self:
> 
> ```cs
> internal static async Task<List<T>> ToListAsync<T>(this IAsyncEnumerable<T> source)
> {
>     var result = new List<T>();
>     await foreach (var item in source.ConfigureAwait(false)) result.Add(item);
>     return result;
> }
> ```
> 
> The `System.Linq.Async` package reference was removed from the core libraries so that everyone using SK wasn't forced to use it as well.

Ahhh I didn't know the package was removed intentionally. Removed here also and added a local helper method.
